error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/AssertJMapRules.java

```diff
@@ -171,13 +176,31 @@ AbstractBooleanAssert<?> before(Map<K, V> map, K key) {
       return assertThat(map.containsKey(key)).isFalse();
     }
 
+    @BeforeTemplate
+    AbstractCollectionAssert<?, Collection<? extends K>, K, ?> before2(Map<K, V> map, K key) {
+      return assertThat(map.keySet()).doesNotContain(key);
+    }
+
     @AfterTemplate
     @UseImportPolicy(STATIC_IMPORT_ALWAYS)
     MapAssert<K, V> after(Map<K, V> map, K key) {
       return assertThat(map).doesNotContainKey(key);
     }
   }
 
+  static final class AssertThatMapContainsOnlyKey<K, V> {
```

Comment:
As compared to the `+ "s"` variant just below this matches a single element, as opposed to an `Iterable`. The class name had to be unique, even though they have the same target method in `after`, which dictates the class name. Due to differing arguments to the templates this had to be a separate class.

Reply:
Refaster does support incompatible parameter types in a way that would allow us to merge these rules, but it's awkward at best. I'll just add a comment to track this topic for another time.